src/InformaticsGateway/ExecutionPlugins/ExternalAppOutgoing.cs

```diff
@@ -0,0 +1,142 @@
+﻿/*
+ * Copyright 2021-2023 MONAI Consortium
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using FellowOakDicom;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Monai.Deploy.InformaticsGateway.Api;
+using Monai.Deploy.InformaticsGateway.Common;
+using Monai.Deploy.InformaticsGateway.Configuration;
+using Monai.Deploy.InformaticsGateway.Database.Api.Repositories;
+using Monai.Deploy.InformaticsGateway.Logging;
+
+namespace Monai.Deploy.InformaticsGateway.ExecutionPlugins
+{
+    [PluginName("Remote App Execution Outgoing")]
+    public class ExternalAppOutgoing : IOutputDataPlugin
+    {
+        private readonly ILogger<ExternalAppOutgoing> _logger;
+        private readonly IServiceScopeFactory _serviceScopeFactory;
+        private readonly PluginConfiguration _options;
+
+        public string Name => "Remote App Execution Outgoing";
+
+        public ExternalAppOutgoing(
+            ILogger<ExternalAppOutgoing> logger,
+            IServiceScopeFactory serviceScopeFactory,
+            IOptions<PluginConfiguration> configuration)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _serviceScopeFactory = serviceScopeFactory ?? throw new ArgumentNullException(nameof(serviceScopeFactory));
+            _options = configuration.Value ?? throw new ArgumentNullException(nameof(configuration));
+            if (_options.RemoteAppConfigurations.ContainsKey("ReplaceTags") is false) { throw new ArgumentNullException(nameof(configuration)); }
+        }
+
+        public async Task<(DicomFile dicomFile, ExportRequestDataMessage exportRequestDataMessage)> Execute(DicomFile dicomFile, ExportRequestDataMessage exportRequestDataMessage)
+        {
+            var tags = IDicomToolkit.GetTagArrayFromStringArray(_options.RemoteAppConfigurations["ReplaceTags"]);
+            var outgoingUid = dicomFile.Dataset.GetString(tags[0]);
```

Comment:
`tag[0]` needs to be documented in case someone puts a value other than `StudyInstanceUid`. Or, maybe we just hard code it here?